families/burrow_evm/src/sawtooth_burrow_evm/common/evm_address.go

```diff
@@ -67,13 +69,13 @@ func NewEvmAddrFromBytes(b []byte) (*EvmAddr, error) {
 }
 
 func NewEvmAddrFromString(s string) (ea *EvmAddr, err error) {
-	defer func() {
-		if r := recover(); r != nil {
-			ea = nil
-			err = fmt.Errorf("%v could not be decoded: %v", s, r)
-		}
-	}()
-	return NewEvmAddrFromBytes(client.MustDecode(s))
+	bytes, err := hex.DecodeString(s)
+	if err != nil {
+		return nil, &processor.InternalError{ Msg: fmt.Sprintf(
```

Comment:
The errors defined in sawtooth_sdk/processor are only for use by a transaction processor (code in `sawtooth_burrow_evm/processor/handler`) so this should just use `fmt.Errorf()` again.